packages/fiori/src/DynamicPageTitle.ts

```diff
@@ -0,0 +1,322 @@
+ * content that is displayed.
+ *
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 1.23.0
+ */
+@customElement({
+	tag: "ui5-dynamic-page-title",
+	fastNavigation: true,
+	renderer: litRender,
+	styles: DynamicPageTitleCss,
+	template: DynamicPageTitleTemplate,
+})
+
+/**
+ * Event is fired when the title is toggled.
+ * @private
+ */
+@event("_toggle-title")
+
+class DynamicPageTitle extends UI5Element {
+	/**
+	 * Defines if the title is snapped.
+	 *
+	 * @public
+	 * @default false
+	 */
+	@property({ type: Boolean })
+	snapped!: boolean;
+
+	/**
+	 * Defines if the mobileNavigationActions are shown.
+	 *
+	 * @private
+	 */
+	@property({ type: Boolean })
+	mobileNavigationActions!: boolean;
+
+	/**
+	 * Indicates if the elements is on focus
+	 * @private
+	 */
+	@property({ type: Boolean })
+	focused!: boolean;
+
+	/**
+	 * Defines the content of the Heading of the Dynamic Page.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	heading!: HTMLElement[];
+
+	/**
+	 * Defines the heading that is shown only when the header is snapped.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	snappedHeading!: HTMLElement[];
+
+	/**
+	 * Defines the heading that is shown only when the header is expanded.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	expandedHeading!: HTMLElement[];
+
+	/**
+	 * Defines the actions in the Dynamic page title.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	actions!: HTMLElement[];
+
+	/**
+	 * Defines the navigation actions in the Dynamic page title.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	navigationActions!: HTMLElement[];
+
+	/**
+	 * Defines the content of the Dynamic page title.
+	 *
+	 * @public
+	 */
+	@slot({ "default": true, type: HTMLElement })
+	content!: HTMLElement[];
+
+	/**
+	 * Defines the content of the title that is shown only when the header is snapped.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	snappedContent!: HTMLElement[];
+
+	/**
+	 * Defines the content of the title that is shown only when the header is expanded.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	expandedContent!: HTMLElement[];
+
+	/**
+	 * Defines the content of the breadcrumbs inside Dynamic Page Title.
+	 *
+	 * @public
+	 */
+	@slot({ type: HTMLElement })
+	breadcrumbs!: HTMLElement[];
+
+	static i18nBundle: I18nBundle;
+
+	_handleResize: ResizeObserverCallback;
+	minContentWidth?: number;
+	minActionsWidth?: number;
+
+	constructor() {
+		super();
+		this._handleResize = this.handleResize.bind(this);
+	}
+
+	static async onDefine() {
+		DynamicPageTitle.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._handleResize);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+	}
+
+	onBeforeRendering() {
+		this.prepareLayoutActions();
+	}
+
+	get classes() {
```

Comment:
classes are unconditional, always present, you can just write them in the template - it becomes more informative as immediately shows which CSS class, which DOM element serves.